src/gridcoin/upgrade.cpp

```diff
@@ -875,13 +884,129 @@ void Upgrade::DeleteSnapshot()
+
+    fs::directory_iterator IterEnd;
+
+    try {
+        for (fs::directory_iterator Iter(cleanup_path); Iter != IterEnd; ++Iter) {
+            if (fs::is_regular_file(*Iter)) {
+                size_t FileLoc = Iter->path().filename().string().find("blk");
+
+                if (FileLoc != std::string::npos) {
+                    std::string filetocheck = Iter->path().filename().string();
+
+                    // Check it ends with .dat and starts with blk
+                    if (filetocheck.substr(0, 3) == "blk" && filetocheck.substr(filetocheck.length() - 4, 4) == ".dat") {
+                        fs::path new_name = *Iter;
+                        new_name.replace_extension(".dat.orig");
+
+                        uintmax_t file_size = fs::file_size(Iter->path());
+
+                        // Rename with orig as the extension, because ProcessBlock will load blocks into a new block data
+                        // file.
+                        fs::rename(*Iter, new_name);
+                        block_data_files.insert(std::make_pair(new_name, file_size));
+                    }
+                }
+            }
+        }
+    } catch (fs::filesystem_error &ex) {
+        error("%s: Exception occurred: %s. Failed to rename block data files to blk*.dat.orig in preparation for "
+              "reindexing.", __func__, ex.what());
+
+        return false;
+    }
+
+    return true;
+}
+
+bool Upgrade::LoadBlockchainData(std::set<std::pair<fs::path, uintmax_t>>& block_data_files, bool cleanup_imported_files)
+{
+    bool successful = true;
+
+    uintmax_t total_size = 0;
+    uintmax_t cumulative_size = 0;
+
+    for (const auto& iter : block_data_files) {
+        total_size += iter.second;
+    }
+
+    if (!total_size) return false;
+
+    try {
+        for (const auto& iter : block_data_files) {
+
+            unsigned int percent_start = cumulative_size * (uintmax_t) 100 / total_size;
+
+            cumulative_size += iter.second;
+
+            unsigned int percent_end = cumulative_size * (uintmax_t) 100 / total_size;
+
+            FILE *block_data_file = fsbridge::fopen(iter.first, "rb");
+
+            LogPrintf("INFO: %s: Loading blocks from %s.", __func__, iter.first.filename().string());
+
+            if (!LoadExternalBlockFile(block_data_file, iter.second, percent_start, percent_end)) {
+                successful = false;
+
+                break;
+            }
+        }
+    } catch (fs::filesystem_error &ex) {
+        error("%s: Exception occurred: %s. Failure occurred during attempt to load blocks from original "
+              "block data file(s).", __func__, ex.what());
+
+        successful = false;
+    }
+
+    if (successful) {
+        // Only delete the source files that were imported if cleanup_imported_files is set to true
+        if (cleanup_imported_files) {
+            try {
+                fs::path cleanup_path;
+
+                if (!GetActualCleanupPath(cleanup_path)) return false;
+
```

Comment:
Seems redundant.

Reply:
Yes. I simplified the approach earlier and forgot to remove those.